compiler/tests-full/stdlib.cma.expected.js

```diff
@@ -17233,385 +17218,426 @@
      /*<<camlinternalFormat.ml:1502:2>>*/ bprint_fmtty(buf, fmtty);
      /*<<camlinternalFormat.ml:1502:24>>*/ return buffer_contents(buf) /*<<camlinternalFormat.ml:1503:21>>*/ ;
    }
-   function make_printf$0(counter, k, acc, fmt){
-    var
-     k$0 =  /*<<camlinternalFormat.ml:1518:17>>*/ k,
-     acc$0 = acc,
-     fmt$0 = fmt;
-    for(;;){
-     if(typeof fmt$0 === "number")
-       /*<<camlinternalFormat.ml:1605:4>>*/ return caml_call1(k$0, acc$0) /*<<camlinternalFormat.ml:1605:9>>*/ ;
-      /*<<camlinternalFormat.ml:1518:17>>*/ switch(fmt$0[0]){
-       case 0:
-        var rest = fmt$0[1];
-         /*<<camlinternalFormat.ml:1520:4>>*/ return function(c){
-         var new_acc =  /*<<camlinternalFormat.ml:1521:6>>*/ [5, acc$0, c];
-          /*<<camlinternalFormat.ml:1522:6>>*/ return make_printf
-                 (k$0, new_acc, rest) /*<<camlinternalFormat.ml:1522:32>>*/ ;} /*<<camlinternalFormat.ml:1520:4>>*/ ;
-       case 1:
-        var rest$0 =  /*<<camlinternalFormat.ml:1518:17>>*/ fmt$0[1];
-         /*<<camlinternalFormat.ml:1524:4>>*/ return function(c){
-         var
-          str =
-             /*<<camlinternalFormat.ml:1493:12>>*/ caml_call1
-             (Stdlib_Char[2], c),
-          l =  /*<<camlinternalFormat.ml:1494:2>>*/ caml_ml_string_length(str),
-          res =
-             /*<<camlinternalFormat.ml:1495:12>>*/ caml_call2
-             (Stdlib_Bytes[1], l + 2 | 0, 39);
-          /*<<camlinternalFormat.ml:1496:2>>*/ caml_blit_string
-          (str, 0, res, 1, l);
+   function make_printf$0(counter, k$2, acc$4, fmt$2){
+    a:
+    {
+     b:
+     {
+      c:
+      {
+       d:
+       {
+        e:
+        {
+         f:
+         {
+          g:
+          {
+           h:
+           {
+            i:
+            {
+             j:
+             {
```

Comment:
Do you understand this?

Reply:
I did not look at it yet

Reply:
This is because we have new tailcalls introducing loops and  `shrink_loops` tries to exit the loop for non-small code-path. Here, I believe the presence of closure is responsible for exiting the loop, we have the following in `Structure.measure`
```
          (* A closure is never small *)
          | Let (_, Closure _) -> -1
```

Reply:
I'm not sure what to do about this

Reply:
So, that's the right thing to do, then.